tools/fuchsia/sdk/component.gni

```diff
@@ -0,0 +1,233 @@
+# Copyright 2019 The Fuchsia Authors. All rights reserved.
+# Use of this source code is governed by a BSD-style license that can be
+# found in the LICENSE file.
+
+import("cmc.gni")
```

Comment:
can we bring in the gn-sdk dep to get some of these rules? Some of them feel like they should be provided by the sdk.

Reply:
So it turns out that the gn-sdk is not really helpful to Flutter.

For context, there are some recent learnings, a Fuchsia SDK design review, and approved design doc (on the Fuchsia side) that resulted in this approach. (The design doc was specific to adding `TestWithEnvironment` to the SDK, but the proposal was the result of the guidance I received, specific to this Flutter integration testing effort, which I've summarized below.):

The gn-sdk was really set up more as a "chromium-sdk". In fact, the gn-sdk is a supposed to be a superset of the core-sdk, but it isn't. Some things are explicitly removed from gn-sdk, including (surprisingly) the Fuchsia Dart APIs! So "upgrading" to gn-sdk resulted in losing those APIs.

When I discussed this with the SDK team, they did not want to add the Dart APIs to the gn-sdk. I expect the SDK's to be restructured in the future, but for now, we want the core-sdk.

Second reason: Even if we had the gn-sdk PLUS the Dart APIs from core, the gn-sdk included only TWO files that I had to manually copy into `src/flutter/tools/fuchsia/`. Most of the other additional to `src/flutter/tools/fuchsia/` come directly from `fuchsia.git` (with some minor tailoring) and were not available via either SDK. 

Also, Fuchsia is not planning to add any more GN rules to _any_ SDK in the future. They don't expect any project other that Flutter is likely to need similar build rules, using GN. So I was asked to use whatever existing support I could find in existing Flutter GN rules, and add (copy from fuchsia.git) whatever was missing. 

As best I could tell, flutter/engine didn't appear to have existing support for Dart-based Fuchsia components, I had to bring in build rules from fuchsia.git.

Due to the complexity of these rules, I preferred to keep the original fuchsia.git build structure and logic as close to the original as possible, which probably resulted in _some_ duplication of existing logic in flutter/engine, but it seemed like the duplication was minimal, and the additional logic accounted for the vast majority of what I had to add.